### ISSUE TITLE
suppress "correlation matrix not shown" message

### ISSUE THREAD

**Issue by bbolker** [lme4/lme4#725]
From Terry Therneau:

> I commonly use `summary(lmerfit, cor=FALSE)` to suppress the correlation matrix.   It is a minor annoyance to have it still print the following lines whenever the number of variables is > 12:

```
Correlation matrix not shown by default, as p = 13 > 12.
Use print(x, correlation=TRUE)  or
    vcov(x)        if you need it
```

> This is minor, perhaps worhtwhile if you are already updating something else.

Terry Therneau

**Comment by bbolker** [comment]
@mmaechler, I think you might be failing to make a distinction between `print.merMod()`/`print.summary.merMod()` and `summary.merMod()` methods ... ?

```r
library(lme4)
set.seed(101)
dd <- data.frame(w = rnorm(1000), x = rnorm(1000), y = rnorm(1000), z = rnorm(1000),
                               f = factor(rep(1:50, each = 20)))
dd$resp <- simulate(~ w*x*y*z + (1|f), newdata = dd,
                                     newparams = list(beta = rep(1, 16), theta = 1, sigma = 1),
                                     family = gaussian)[[1]]
m1 <- lmer(resp ~ w*x*y*z + (1|f), dd)
summary(m1, cor = FALSE)  ## prints message
print(summary(m1, cor = FALSE), cor = FALSE) ## doesn't print message
```

**Comment by mmaechler** [comment]
Uhm, yes, you are right: there is an issue that I overlooked.
Still I don't like your proposed solution:   for largish 'p' (not so common in ecology maybe, but not unrealistic),
 I'd not be happy to compute this by default every time, but never make use of it (by default)... and then also store it w/o using it.. --> user will more commonly (like Terry)  have to say  `summary(*,  correlation=FALSE)` routinely to prevent the "non-sense".

I'd prefer a slightly more slick solution: store an extra attribute with the summary to indicate "no correlation explicitly", and use it in   `print.summary.merMod()` ..
I can see that this "works" (for some version of "works", however  `summary(*, correlation=TRUE)` now (in my uncommitted version) does *not* print the message anymore (and also does not print the correlation matrix)

*However* this is still not "intuitive" and useful....
and it is *still* the case that the help page part I mentioned is crucial:
The correlation matrix is *never* printed unless  `print(<summary.merMod>, correlation=TRUE)` is used.
It does *not* play any role if someone used  `summary(*, correlation = NA/TRUE/FALSE)` : The printing of the correlation matrix is entirely activated by the *print()* method `correlation=TRUE/FALSE` and not.

Terry's "wrong expectation" really comes from the fact that we have a different (more flexible but hence more complicated) logic than `lm()`.  In that case, indeed,  `summary(*, correlation=TRUE)` means that the correlation matrix is computed and stored with the object and then it is always printed by the print method.
We wanted to be able to tell it to compute the correlation matrix and then still allow the user to not print it ... which I think is really good.  What is *not* good I now think is that over time we wanted too much flexibility and ended up in a situation that is inconvenient  and confusing for those who expect  lm()-like behavior.

I think we should *also* provide that, i.e. the *default* of the `correlation` argument of the print() method should become more sophisticated...  I'll see if that seems fulfillable.

**Comment by mmaechler** [comment]
fixed with da657fb

Here is "test code" to cover all cases:
```
set.seed(101)
dd <- data.frame(w = rnorm(250), x = rnorm(250), y = rnorm(250), z = rnorm(250),
                 f = factor(rep(1:50, each = 5)))
dd$resp <- simulate(~ w*x*y*z + (1|f), newdata = dd,
                    newparams = list(beta = rep(1, 16), theta = 1, sigma = 1),
                    family = gaussian)[[1]]
m0 <- lmer(resp ~ w*x     + (1|f), dd) #  small  p ==> prints corr by default
m1 <- lmer(resp ~ w*x*y*z + (1|f), dd) # "large" p ==>    no corr  by default

   (sm0  <- summary(m0))             ## prints corr
   (sm0T <- summary(m0, cor=TRUE ))  ## prints corr
   (sm0F <- summary(m0, cor=FALSE))  ## does *not* print corr *nor* msg
print(sm0,    cor= TRUE) ## prints corr
print(sm0T,   cor= TRUE) ## prints corr
print(sm0F,   cor= TRUE) ## prints corr *after* msg ".. could have been required .."
print(sm0,    cor=FALSE) #  no corr nor msg
print(sm0T,   cor=FALSE) ## no corr nor msg
print(sm0F,   cor=FALSE) ## no corr nor msg

   (sm1  <- summary(m1))               ## prints message
   (sm1F <- summary(m1, cor = FALSE))  ## no corr no message
   print(sm1F)                         ## (the same)
   (sm1T <- summary(m1, cor = TRUE))   ## now *does* print corr

print(sm1,    cor =  TRUE) # prints corr {+extra msg: "could have been required"}
print(sm1T,   cor =  TRUE) # prints corr (no msg)
print(sm1F,   cor =  TRUE) # prints corr {+extra msg: "could have been required"}
print(sm1,    cor = FALSE) # no corr nor msg
print(sm1T,   cor = FALSE) # no corr nor msg
print(sm1F,   cor = FALSE) # no corr nor msg

## and indeed, all differ
all.equal(sm1, sm1T) # 2 'vcov' "factors"  +  "corrSet"
all.equal(sm1, sm1F) # "corrSet" only
all.equal(sm1T, sm1F)# 2 'vcov' "factors"  +  "corrSet"
```
